### PR TITLE
New

### PR DESCRIPTION
**Pull request by thelightreaper123**

### CODE REVIEW

File: week6/src/App.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React, {useState, useEffect} from 'react';
+import './App.css';
+import List from './List';
+import Button from "./Button";
+import { BrowserRouter as Router, Route, Switch, createBrowserRouter, RouterProvider, createRoutesFromElements, Link, useNavigate} from 'react-router-dom';
+import{FontAwesomeIcon} from '@fortawesome/react-fontawesome'
+
+
+function App() {
+  const storedTask = JSON.parse(localStorage.getItem('items')) || []
```
Comment (tayoogunbiyi) [inline week6/src/App.js:10]: 👍

**Review by tayoogunbiyi — APPROVED** [review]

File: week6/src/App.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ function App() {
       items.map(item=>{
       return(
         <div>
-          <List item={item} deleteit = {deleteitem}/><Link to={`/task/${item.id}`}><p>EXPAND</p></Link>
+          <List item={item} deleteit = {deleteitem}/><Link to={`/${item.value}`}><p>EXPAND</p></Link>
```
Comment (tayoogunbiyi) [inline week6/src/App.js:61]: q: why did you switch from id to value?

File: week6/src/Task.js [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+      return(
+        <div>
+           <h1>404</h1>
+        <h2>PAGE NOT FOUND</h2>
+        <p>it appears the task you are looking for does not exist</p>
+        <p>Click link to go back to <Link to='/'>HomePage</Link></p>
+        </div>
+       );
+    }
   })
```
Comment (tayoogunbiyi) [inline week6/src/Task.js]: This doesn't work as expected. See the screenshot below

Screenshot 2023-07-31 at 08 49 13

**Review by tayoogunbiyi — CHANGES_REQUESTED** [review]
Please address the changes

**Review by tayoogunbiyi — APPROVED** [review]

File: week6/src/CreateTask.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      alert("please input task!");
+      return;
+    }
+    if (!descriptions){
+        alert("please input description!");
+        return;
+      }
+    const taskinput = /^(?=.*[a-z])/;
+    if (!taskinput.test(tasks)){
+      alert('must contain at least one letter');
```
Comment (tayoogunbiyi) [inline week6/src/CreateTask.js:33]: The alert says 'must contain at least ....' but does not make it clear that it is referred to the task name? 
Same thing as description below

**Review by tayoogunbiyi — CHANGES_REQUESTED** [review]

File: week6/src/CreateTask.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      return;
+    }
+    const descriptionInput = /^(?=.*[a-z])/;
+      if (!descriptionInput.test(descriptions)){
+        alert('must contain at least one letter');
+        return;
+      }
+
```
[the remark this answers is not in the document]
Comment (thelightreaper123) [reply week6/src/CreateTask.js:41]: I ran into an error when they were separated and moving it together fixed it

File: week6/src/App.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import './App.css';
+import CreateTask from './CreateTask';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+
+function App() {
+   const [modal, setModal] = useState(false);
+   const toggle=()=>{
+    setModal(!modal);
+   }
```
[the remark this answers is not in the document]
Comment (tayoogunbiyi) [reply week6/src/App.js:11]: you didn't address this

File: week6/src/App.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <>
+      <div className='Main'>
+    <div className='header'>
+   <h1>Tades Task App</h1>
+   <button className='create-task-btn' onClick={()=>setModal(true)}>Create task</button>
+   </div>
+   <div className='body'>
+
+   <CreateTask toggle={toggle} modal={modal}/>
```
[the remark this answers is not in the document]
Comment (tayoogunbiyi) [reply week6/src/App.js:21]: neither did you address this

File: week6/src/CreateTask.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    const newArray1 = items.filter(item => item.id !== id);
+    const newArray2 = narration.filter(description => description.id !== id);
+    const newArray3 = date.filter(date => date.id !== id);
+    setItems(newArray1);
+    setDescription(newArray2);
+    setDate(newArray3);
+     } 
+  return (
+    <div>
+      <Modal isOpen={modal} toggle={toggle}>
```
Comment (tayoogunbiyi) [inline week6/src/CreateTask.js:97]: you can see that 
1. the component is called Modal
2. you are passing a prop value called modal too. Could we name is isOpen instead? Apply this same process to the other places we use modal wrongly

File: week6/src/CreateTask.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        <ModalBody>
+          <form>
+          <div className='form-group'>
+          <label>Task Name</label>
+          <input placeholder='input task' type='text' className='form-control' value={tasks} onChange={e => setNewitem(e.target.value)}/>
+          </div>
+          <label>Description</label>
+          <textarea rows="5" className='form-control' value={descriptions} onChange={e => setNewDescription(e.target.value)}></textarea>
+          <label>Date created</label>
+          <textarea rows="2" className='form-control' value={time} onChange={e => setNewDate(e.target.value)}></textarea>
```
Comment (tayoogunbiyi) [inline week6/src/CreateTask.js:108]: date shouldn't be a plain textarea. we can use a date picker instead - https://www.w3schools.com/tags/att_input_type_date.asp

Comment (tayoogunbiyi) [reply week6/src/CreateTask.js:108]: that will help with properly validating the date too

**Review by tayoogunbiyi — CHANGES_REQUESTED** [review]

**Review by tayoogunbiyi — APPROVED** [review]

**Pull request merged**